protoc-gen-kotlin/lib/src/commonMain/kotlin/pbandk/gen/CodeGenerator.kt

```diff
@@ -26,7 +30,35 @@ open class CodeGenerator(
         fn().also { indent = indent.dropLast(4) }
     }
 
-    protected fun writeType(type: File.Type, parentType: String? = null) = when (type) {
+    private fun writeExtension(field: File.Field) {
+        when (field) {
+            is File.Field.Numbered -> {
+                val localType = when (field.type) {
+                    File.Field.Type.MESSAGE -> ((field as File.Field.Numbered.Standard).localTypeName ?: field.kotlinLocalTypeName)?.substringAfter(".")
```

Comment:
What if the field is `File.Field.Numbered.Wrapper`? I think you need to handle that case too.